runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -177,65 +177,72 @@ public long append(Object object,
                        Function<TokenResponse, Boolean> deacquisitionCallback) {
         final LogData ld = new LogData(DataType.DATA, object, runtime.getParameters().getCodecType());
 
-        // Validate if the  size of the log data is under max write size.
-        ld.checkMaxWriteSize(runtime.getParameters().getMaxWriteSize());
-
-        // First, we get a token from the sequencer.
-        TokenResponse tokenResponse = runtime.getSequencerView()
-                .next(id);
-
-        // We loop forever until we are interrupted, since we may have to
-        // acquire an address several times until we are successful.
-        for (int x = 0; x < runtime.getParameters().getWriteRetry(); x++) {
-            // Next, we call the acquisitionCallback, if present, informing
-            // the client of the token that we acquired.
-            if (acquisitionCallback != null) {
-                if (!acquisitionCallback.apply(tokenResponse)) {
-                    // The client did not like our token, so we end here.
-                    // We'll leave the hole to be filled by the client or
-                    // someone else.
-                    log.debug("Acquisition rejected token={}", tokenResponse);
-                    return -1L;
+        // Opening serialization handle before acquiring token, this way we prevent the
+        // readers to wait for the possibly long serialization time in writer.
+        // The serialization here only serializes the payload because the token is not
+        // acquired yet, thus metadata is incomplete. Once a token is acquired, the
+        // writer will append the serialized metadata to the buffer.
+        try (ILogData.SerializationHandle sh = ld.getSerializedForm(false)) {
+            // Validate if the  size of the log data is under max write size.
+            int payloadSize = ld.checkMaxWriteSize(runtime.getParameters().getMaxWriteSize());
```

Comment:
Instead of having `checkMaxWriteSize` return the size, you can directly access it from the serialization handle